src/phoenix/datasets/evaluators.py

```diff
@@ -192,6 +192,75 @@ def _parse_label_from_explanation(raw_string: str) -> str:
     return raw_string
 
 
+class RelevanceEvaluator:
+    annotator_kind = "LLM"
+    template = (
+        "Determine if the following response is relevant to the query. In this context, "
+        "'relevance' means that the response directly addresses the core question or topic of the "
+        "query. First, explain step-by-step why you think the text is or is not relevant. "
+        "Then provide a single word label; 'true' if the text is relevant or 'false' if the text "
+        "is not relevant. "
+        "Here is an example template for your reponse:\n\n"
+        "CRITERIA: the response is 'relevant' to the query\n"
+        "QUERY: *text that contains a query*\n"
+        "RESPONSE: *a response that may or may not be relevant to the query*\n"
+        "EXPLANATION: *a step by step explanation of your reasoning for whether or not the "
+        "response is relevant to the query*\n"
+        "LABEL: *true or false*\n\n"
+        "Follow this template for the following example:\n\n"
+        "CRITERIA: the response is 'relevant' to the query\n"
+        "QUERY: {reference}\n"
+        "RESPONSE: {submission}\n"
+        "EXPLANATION: "
+    )
+
+    def __init__(
+        self,
+        model: LLMBaseModel,
+        query_fn: Callable[[Example, ExperimentRun], str],
+        response_fn: Callable[[Example, ExperimentRun], str],
+        name: str = "RAGRelevanceEvaluator",
+    ):
+        self.model = model
+        self.name = name
+        self.query_fn = query_fn
+        self.response_fn = response_fn
+
+    def _format_eval_template(self, example: Example, experiment_run: ExperimentRun) -> str:
+        assert experiment_run.output is not None
+        query = self.query_fn(example, experiment_run)
+        response = self.response_fn(example, experiment_run)
```

Comment:
Should `response_fn` take in the `query` as an argument? I am assuming `response_fn` will contain the chain, query engine, agent, etc. that generates the response given a query.

Reply:
Actually, this looks like it's mainly intended for extracting values.